addons/website_forum/models/forum.py

```diff
@@ -844,23 +845,22 @@ def convert_comment_to_answer(self, message_id, default=None):
         return new_post
 
     def unlink_comment(self, message_id):
-        result = []
-        for post in self:
-            user = self.env.user
-            comment = self.env['mail.message'].sudo().browse(message_id)
-            if not comment.model == 'forum.post' or not comment.res_id == post.id:
-                result.append(False)
-                continue
-            # karma-based action check: must check the message's author to know if own or all
-            karma_unlink = (
-                comment.author_id.id == user.partner_id.id and
-                post.forum_id.karma_comment_unlink_own or post.forum_id.karma_comment_unlink_all
-            )
-            can_unlink = user.karma >= karma_unlink
-            if not can_unlink:
-                raise AccessError(_('%d karma required to unlink a comment.', karma_unlink))
-            result.append(comment.unlink())
-        return result
+        user = self.env.user
+        comment = self.env['mail.message'].sudo().browse(message_id)
+        if comment.model != 'forum.post' or comment.res_id != self.id:
+            return False
+
+        # karma-based action check: must check the message's author to know if own or all
+        if comment.author_id == user.partner_id:
+            karma_unlink = self.forum_id.karma_comment_unlink_own
+        else:
+            karma_unlink = self.forum_id.karma_comment_unlink_all
+
+        if user.karma < karma_unlink:
+            raise AccessError(_('%d karma required to unlink a comment.', karma_unlink))
+
+        comment.unlink()
```

Comment:
Is it really working ? I mean, most people don't have unlink access on mail.message model ... normally 8-) .

Reply:
`comment` is sudo so it will always delete.